admin_guide/advanced_scheduling/taints_tolerations.adoc

```diff
@@ -0,0 +1,401 @@
+----
+
+[[admin-guide-taints-daemonsets]]
+== Daemonsets and Tolerations
+
+link:https://kubernetes.io/docs/admin/daemons/[DaemonSet] pods are created with `NoExecute` tolerations for `node.alpha.kubernetes.io/unreachable` and `node.alpha.kubernetes.io/notReady`
+with no `tolerationSeconds` to ensure that DaemonSet pods are never evicted due to these problems, even when when the Default Toleration Seconds feature is disabled.
+
+[[admin-guide-taints-use-cases]]
+== Example Use Cases
+
+Taints and tolerations are a flexible way to steer pods away from nodes or evict pods that should not be running on a node. A few of the use cases are:
+
+* xref:admin-guide-taints-dedicated-user[Dedicating a node for a user]
+* xref:admin-guide-taints-dedicated-only[Binding a user to a node]
+* xref:admin-guide-taints-special[Dedicating nodes with special hardware]
+
+[[admin-guide-taints-dedicated-user]]
+=== Dedicating a Node for a User
+
+You can specify a set of nodes for exclusive use by a particular set of users.
+
+To specify dedicated nodes:
+
+. Add a taint to those nodes:
++
+For example:
++
+----
+$ oadm taint nodes node1 dedicated=groupName:NoSchedule`
+----
+
+. Add a corresponding toleration to the pods. This would be done most easily by writing a custom xref:../install_config/master_node_configuration.adoc#master-config-admission-control-config[admission controller].
++
+Only the pods with the tolerations will be allowed to use the dedicated nodes.
+
+[[admin-guide-taints-dedicated-only]]
+=== Binding a User to a Node
+
+You can configure a node so that particular users can use only the dedicated nodes.
+
+To configure a node so that users can use only that node:
+
+. Add a taint to those nodes:
++
+For example:
++
+----
+$ oadm taint nodes node1 dedicated=groupName:NoSchedule`
+----
+
+. Add a corresponding toleration to the pods. This would be done most easily by writing a custom xref:../install_config/master_node_configuration.adoc#master-config-admission-control-config[admission controller].
++
+The admission controller should add a node affinity to require that the pods can only schedule onto nodes labeled with the `key:value` label (`dedicated=groupName`).
+
+. Add a label similar to the taint (such as the `key:value` label) to the dedicated nodes.
+
+
+[[admin-guide-taints-special]]
+=== Nodes with Special Hardware
+
+In a cluster where a small subset of nodes have specialized hardware (for example GPUs), you can use taints and tolerations to keep pods that do not need the specialized hardware off of those nodes, leaving the nodes for pods that do need the specialized hardware. You can also require pods that need specialized hardware to use specific nodes.
+
+To make sure pods are blocked from the specialized hardware:
+
+. Taint the nodes that have the specialized hardware using one of the following commands:
++
+----
+$ oadm taint nodes <node-name> special=true:NoSchedule
+$ oadm taint nodes <node-name> special=true:PreferNoSchedule
```

Comment:
I am not suggesting use "true/false/yes/no" as value, since those string-like values will be translated / treated as "1/0" in pod yaml file.
I prefer disktype=ssd or something like this.